File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/HabitDraft.kt
package com.goalmaker.app.application.planning

import java.time.LocalDate

/**
 * A habit as the owner typed it, before [HabitList] cleans it: the fields that don't fit the cadence
 * or the measure are dropped there (docs/habits.md).
 */
data class HabitDraft(
    val name: String,
    val startsOn: LocalDate,
    val cadence: String = HabitRules.DAILY,
    val weekdays: Int? = null,
    val times: Int? = null,
    val measure: String = HabitRules.CHECK,
    val target: Double? = null,
    val unit: String? = null,
    val emoji: String? = null,
    val goalId: String? = null,
)

File: windows/src/GoalMaker.Core/Planning/HabitDraft.cs
namespace GoalMaker.Core.Planning;

/// <summary>
/// A habit as the owner typed it, before <see cref="HabitList"/> cleans it: the fields that don't fit
/// the cadence or the measure are dropped there (docs/habits.md).
/// </summary>
public sealed record HabitDraft(string Name, DateOnly StartsOn)
{
    public string Cadence { get; init; } = HabitRules.Daily;

    public int? Weekdays { get; init; }

    public int? Times { get; init; }

    public string Measure { get; init; } = HabitRules.Check;

    public double? Target { get; init; }

    public string? Unit { get; init; }

    public string? Emoji { get; init; }

    public string? GoalId { get; init; }
}
